smarts/core/agent.py

```diff
@@ -63,9 +63,10 @@ def act(self, obs):
 class AgentPolicy(Agent):
     # we cannot use debtcollector here to signal deprecation because the wrapper object is
     # not pickleable, instead we simply print.
-    logger.warning(
-        "[DEPRECATED] AgentPolicy has been replaced with `smarts.core.agent.Agent`"
-    )
+    def __init__(self):
```

Comment:
:+1: The `DEPRECATED` log is no longer dupplicated?

Reply:
One more suggestion, remove the `{self}` of line 126, 131, 138 to make the log less annoying.

Reply:
Alternate suggestion is to put the `{self}`s in `debug` logging.

Reply:
It is deprecated still, but the code was in the wrong spot.